src/RA_StringUtils.h

```diff
@@ -284,12 +284,15 @@ class StringBuilder
                 oss << arg;
             else
             {
-                if (sFormat.front() == '0')
-                    oss << std::setfill('0');
-                const int nDigits = std::stoi(sFormat);
-                if (nDigits > 0)
-                    oss << std::setw(nDigits);
+                if(std::isdigit(ra::to_unsigned(sFormat.front())))
+                {
+                    if (sFormat.front() == '0')
+                        oss << std::setfill('0');
 
+                    const int nDigits = std::stoi(sFormat);
+                    if (nDigits > 0)
+                        oss << std::setw(nDigits);
+                }
```

Comment:
For a straight-up %X or %x to work, this needed to be done. The unit tests still pass for the ones that were there, I added to the tests a little bit to test the added functionality.